aws/resource_aws_guardduty_publishing_destination.go

```diff
@@ -0,0 +1,182 @@
+				Required: true,
+				ForceNew: true,
+			},
+			"destination_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  guardduty.DestinationTypeS3,
+				ValidateFunc: validation.StringInSlice([]string{
+					guardduty.DestinationTypeS3,
+				}, false),
+			},
+			"destination_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateArn,
+			},
+			"kms_key_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateArn,
+			},
+		},
+	}
+}
+
+func resourceAwsGuardDutyPublishingDestinationCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).guarddutyconn
+
+	detectorID := d.Get("detector_id").(string)
+	input := guardduty.CreatePublishingDestinationInput{
+		DetectorId: aws.String(detectorID),
+		DestinationProperties: &guardduty.DestinationProperties{
+			DestinationArn: aws.String(d.Get("destination_arn").(string)),
+			KmsKeyArn:      aws.String(d.Get("kms_key_arn").(string)),
+		},
+		DestinationType: aws.String(d.Get("destination_type").(string)),
+	}
+
+	log.Printf("[DEBUG] Creating GuardDuty publishing destination: %s", input)
+	output, err := conn.CreatePublishingDestination(&input)
+	if err != nil {
+		return fmt.Errorf("Creating GuardDuty publishing destination failed: %w", err)
+	}
+
+	_, err = waiter.PublishingDestinationCreated(conn, *output.DestinationId, detectorID)
+
+	if err != nil {
+		return fmt.Errorf("Error waiting for GuardDuty PublishingDestination status to be \"%s\": %w",
+			guardduty.PublishingStatusPublishing, err)
+	}
+
+	d.SetId(fmt.Sprintf("%s:%s", d.Get("detector_id"), aws.StringValue(output.DestinationId)))
+
+	return resourceAwsGuardDutyPublishingDestinationRead(d, meta)
+}
+
+func resourceAwsGuardDutyPublishingDestinationRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).guarddutyconn
+
+	destinationId, detectorId, errStateRead := decodeGuardDutyPublishDestinationID(d.Id())
+
+	if errStateRead != nil {
+		return errStateRead
+	}
+
+	input := &guardduty.DescribePublishingDestinationInput{
+		DetectorId:    aws.String(detectorId),
+		DestinationId: aws.String(destinationId),
+	}
+
+	log.Printf("[DEBUG] Reading GuardDuty publishing destination: %s", input)
+	gdo, err := conn.DescribePublishingDestination(input)
+	if err != nil {
+		if isAWSErr(err, guardduty.ErrCodeBadRequestException, "The request is rejected because the one or more input parameters have invalid values.") {
+			log.Printf("[WARN] GuardDuty publishing destination: %q not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Reading GuardDuty publishing destination: '%s' failed: %w", d.Id(), err)
+	}
+
+	d.Set("detector_id", detectorId)
+	d.Set("destination_type", gdo.DestinationType)
+	d.Set("kms_key_arn", gdo.DestinationProperties.KmsKeyArn)
+	d.Set("destination_arn", gdo.DestinationProperties.DestinationArn)
+	d.Set("status", gdo.Status)
```

Comment:
The `status` attribute does not exist in the resource schema, so this will panic in the SDK v2 acceptance testing. Let's just remove this here for now.

```suggestion
```